internal/migrate/versions.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package migrate
+
+import (
+	"database/sql"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+)
+
+/*
+
+Scenarios it doesn't handle properly.
+
+1. Rollback to initial state on multi-step migration.
+
+	Let's say there's a scenario where we run migration steps:
+	1. update a table schema
+	2. move files
+	3. update a table schema
+	4. update a table schema, which fails
+
+	In this case there's no easy way to rollback the moving of files.
+
+2. Undoing migrations.
+
+	Intentionally left out, because we do not gain that much from currently.
+
+3. Snapshotting the whole state.
+
+	This probably should be done by the user of this library, when there's disk-space available.
+
+4. Figuring out what the exact executed steps are.
+*/
+
+// Migration describes a migration steps
+type Migration struct {
+	Table    string
+	OnCreate Action
+	Steps    []*Step
+}
+
+// Step describes a single step in migration.
+type Step struct {
+	Description string
+	Version     int // Versions should start at 1
```

Comment:
if Migrations.Steps is an ordered list, do we really need Step.Version?

Reply:
Yes, it should be ordered. I prefer to keep the version explicit, it's easier to find later the appropriate step.

Reply:
we can add validation to be sure thats ordered.

Reply:
https://golang.org/pkg/sort/#SliceIsSorted

Reply:
I figured the Description handled finding the step later

Reply:
what do you mean?

Reply:
"I prefer to keep the version explicit, it's easier to find later the appropriate step." ... If we see a failure during migrate, we could debug the error based on the friendly 'Description'.  I still don't understand the value of Version.  Its not a big deal though, it just seems like it will cause slightly more merge issues to have it.

Reply:
but how you want to distinguish which steps were applied with older bineries? let say your current db is in version 5 and binaries 10. how to avoid applying 1-5?

Reply:
You just use the index into the Migrate.Steps slice as the version that you store in the database.  The database still says you're at step 5, you just didn't have to explicitly specify Version=5 on the fifth step.  It's definitely nitpicking at this point.